### PR TITLE
[virtual-frameworks] Fix issue with Sourcekit clang

### PR DESCRIPTION
**Pull request by jerrymarino**
This fixes 2 issue which manifest under certain clang files. We
ended up with the argument used for the swift frontend here.
Additionally, it was unintentionally excluding some hmaps from the
invocation.

### CODE REVIEW

**Review by congt — APPROVED** [review]

File: rules/xcodeproj.bzl
Comment (thiagohmcruz) [inline rules/xcodeproj.bzl:92]: So we're assuming that only one `dep` will hold the vfs provider right? Is that always the case or is it possible that more than one `dep` could be holding this provider?

**Review by thiagohmcruz — APPROVED** [review]
I have one small question but LGTM!

File: rules/xcodeproj.bzl
Comment (jerrymarino) [reply rules/xcodeproj.bzl:92]: Yeah exactly, the existing code did that so was going off that 👍

**Pull request merged**
